source/simulator/solver_schemes.cc

```diff
@@ -694,25 +700,30 @@ namespace aspect
                parameters.max_nonlinear_iterations_in_prerefinement)
       :
       parameters.max_nonlinear_iterations;
-    do
+
+    SolverControl nonlinear_solver_control(max_nonlinear_iterations,
+                                           parameters.nonlinear_tolerance,
+                                           true,
+                                           true);
```

Comment:
I would remove the two bool args, the defaults are fine (here and in all other places).